tests/pyoozie/test_builder.py

```diff
@@ -130,12 +119,55 @@ def test_coordinator_submission_xml_with_configuration(username, coord_app_path)
     </configuration>''') == tests.utils.xml_to_dict_unordered(actual)
 
 
-def test_workflow_builder(tmpdir, workflow_builder):
-    with open('tests/data/workflow.xml', 'r') as fh:
-        expected_xml = fh.read()
+@pytest.mark.xfail
+def test_workflow_builder(tmpdir):
+
+    extract_action = tags.Shell(exec_command='echo', arguments=["'Extract data from an operational system'"])
+    transform_action = tags.Shell(exec_command='echo', arguments=["'Transform data'"])
+    load_action = tags.Shell(exec_command='echo', arguments=["'Load data into a database'"])
+
+    workflow_builder = builder.WorkflowBuilder(
+        name='descriptive-name',
+        job_tracker='job-tracker',
+        name_node='name-node',
+    ).add_action(
+        name='extract',
+        action=extract_action,
+        action_on_error=tags.Email(to='person@example.com', subject='Error',
+                                   body='A bad thing happened while extracting'),
+        kill_on_error='Failure message on extracting',
+    ).add_action(
+        name='transform',
+        action=transform_action,
+        action_on_error=tags.Email(to='person@example.com', subject='Error',
+                                   body='A bad thing happened while transforming'),
+        kill_on_error='Failure message on transforming',
+        depends_upon=(extract_action,),
```

Comment:
Actions are added one at a time with the assumption that an dependencies have already been added (if not, an assertion is thrown).